vrealize_operations_integration_sdk/validation/highlights.py

```diff
@@ -0,0 +1,98 @@
+from vrealize_operations_integration_sdk.collection_statistics import LongCollectionStatistics
+from vrealize_operations_integration_sdk.stats import get_growth_rate
+from vrealize_operations_integration_sdk.validation.result import Result
+
+
+def highlight_object_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_growth = [(object_type, stats.objects_growth_rate) for
+                           object_type, stats in
+                           long_collection_statistics.long_object_type_statistics.items()
+                           if stats.objects_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_growth):
+        for obj_type, growth in objects_with_growth:
+            highlights.with_warning(f"Object of type {obj_type} displayed growth of {growth:.2f}%")
+
+    return highlights
+
+
+def highlight_metric_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_metrics_growth = [(object_type, stats.metrics_growth_rate) for
+                                   object_type, stats in
+                                   long_collection_statistics.long_object_type_statistics.items()
+                                   if stats.metrics_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_metrics_growth):
+        for obj_type, growth in objects_with_metrics_growth:
+            highlights.with_warning(f"Object of type {obj_type} displayed metric growth of {growth:.2f}%")
+
+    return highlights
+
+
+def highlight_property_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_property_growth = [(object_type, stats.properties_growth_rate) for
+                                    object_type, stats in
+                                    long_collection_statistics.long_object_type_statistics.items()
+                                    if stats.properties_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_property_growth):
+        for obj_type, growth in objects_with_property_growth:
+            highlights.with_warning(f"Object of type {obj_type} displayed property growth of {growth:.2f}%")
+
+    return highlights
+
+
+def highlight_property_value_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_property_value_growth = [(object_type, stats.property_values_growth_rate) for
+                                          object_type, stats in
+                                          long_collection_statistics.long_object_type_statistics.items()
+                                          if stats.property_values_growth_rate > 0]
+
+    MAX_NUMBER_OF_PROPERTY_VALUES = 256  # This is mostly subjective.
```

Comment:
That is probably too high. This actually might be a case where it's useful to use the number of collections to determine the threshold, something like `min(32, number_of_collections)` -- that is, if we see a new value every collection that's probably a red flag even if there are only a few collections. We've very rarely (if ever) used an enum with more than 32 values, and even when we have they shouldn't change often enough to trigger anything. Usually a few values are very common and the rest are exceptional cases.

Reply:
Thinking about this some more, there may also be some dependency on the number of objects the property exists on as well. For example, we might have an OS property that is slightly different for (because it has a build number, for example, or maybe some hardware identifier), but never changes. That should be OK as well.